src/mlpack/core/cereal/main.cc

```diff
@@ -0,0 +1,35 @@
+#include "pointer_wrapper.hpp"
+#include <fstream>
+
+void ref_to_pointer(double*& p)
+{
+  if(p == nullptr) {
+    p = new double;
+  }
+  std::cout << "address of  pointer" << p << std::endl;
+  
+}
+
+int main(int argc, char* argv[])
+{
+
+  std::ifstream is("data.xml");
+  cereal::XMLInputArchive archive(is);
+  
+  // std::ofstream is("data.xml");
+  // cereal::XMLOutputArchive archive(is);
+  
+  double data = 40;
+  double* d = nullptr;
```

Comment:
Maybe you already do this when serializing, but to make this properly serialize the value, in addition to uncommenting the ofstream/XMLOutputArchive above, I had to set `double* d = &data`. :+1:

Reply:
@rcurtin I agree, It can work when initializing the pointer, but I wanted to make it work even if it is not initialized since I am not sure that it has been initialized everywhere in mlpack